zcnbridge/http/rest.go

```diff
@@ -9,51 +9,74 @@ import (
 	"net/http"
 	"net/url"
 
+	"github.com/0chain/gosdk/core/logger"
 	"github.com/0chain/gosdk/core/util"
 	"github.com/0chain/gosdk/zcnbridge/errors"
-	"github.com/0chain/gosdk/zcnbridge/log"
 	"github.com/0chain/gosdk/zcncore"
 )
 
 const (
 	// SCRestAPIPrefix represents base URL path to execute smart contract rest points.
-	SCRestAPIPrefix      = "v1/screst/"
-	SmartContractAddress = `6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7`
-	RestPrefix           = SCRestAPIPrefix + SmartContractAddress
-	GetAuthorizersPath   = "/getAuthorizerNodes"
+	SCRestAPIPrefix        = "v1/screst/"
+	SmartContractAddress   = `6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7`
+	RestPrefix             = SCRestAPIPrefix + SmartContractAddress
+	PathGetAuthorizerNodes = "/getAuthorizerNodes"
+	PathGetGlobalConfig    = "/getGlobalConfig"
+	PathGetAuthorizer      = "/getAuthorizer"
 )
 
+type Params map[string]string
+
+var Logger logger.Logger
+var defaultLogLevel = logger.DEBUG
+
+func init() {
+	Logger.Init(defaultLogLevel, "0chain-zcnbridge-sdk")
+}
+
 // MakeSCRestAPICall calls smart contract with provided address
 // and makes retryable request to smart contract resource with provided relative path using params.
-func MakeSCRestAPICall(relativePath string, params map[string]string) ([]byte, error) {
+func MakeSCRestAPICall(opCode int, relativePath string, params Params, cb zcncore.GetInfoCallback) {
 	var (
 		resMaxCounterBody []byte
 		hashMaxCounter    int
+		msg               string
 		hashCounters      = make(map[string]int)
 		sharders          = extractSharders()
-		lastErrMsg        string
 	)
 
+	result := make(chan *http.Response)
+	defer close(result)
+
+	var client = NewRetryableClient()
+
 	for _, sharder := range sharders {
-		var (
-			client = NewRetryableClient()
-			u      = makeURL(params, sharder, relativePath)
-		)
+		go func(sharderUrl string) {
+			var u = makeURL(params, sharderUrl, relativePath)
+			Logger.Info(fmt.Sprintf("Query %s", u.String()))
+			resp, err := client.Get(u.String())
+			if err != nil {
+				msg := fmt.Sprintf("%s: error while requesting sharders: %v", sharderUrl, err)
+				Logger.Error(msg)
+				return
+			}
+			result <- resp
+		}(sharder)
+	}
+
+	for range sharders {
+		resp := <-result
```

Comment:
any request to the sharders failed above would make this channel read stuck. Please check  here: https://github.com/0chain/0chain/blob/56e7411ebbfeaf61a03284e66de0e8cd19d08de7/code/go/0chain.net/chaincore/httpclientutil/http_client_util.go#L332 to adjust appropriately.

Reply:
updated